adm_program/modules/userrelations/userrelations_new.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ ***********************************************************************************************
+ * Create user relations
+ *
+ * @copyright 2004-2016 The Admidio Team
+ * @see http://www.admidio.org/
+ * @license https://www.gnu.org/licenses/gpl-2.0.html GNU General Public License v2.0 only
+ *
+ * Parameters:
+ *
+ * usr_id : user id of the first user in the new relation
+ ***********************************************************************************************
+ */
+require_once('../../system/common.php');
+require_once('../../system/login_valid.php');
+
+// Initialize and check the parameters
+$getUsrId = admFuncVariableIsValid($_GET, 'usr_id', 'int');
+
+if ($gPreferences['members_enable_user_relations'] == 0)
+{
+    $gMessage->show($gL10n->get('SYS_MODULE_DISABLED'));
+    // => EXIT
+}
+
+if($getUsrId <= 0)
+{
+    $gMessage->show($gL10n->get('SYS_NO_ENTRY'));
+    // => EXIT
+}
+
+$user = new User($gDb, $gProfileFields, $getUsrId);
+
+if($user->isNewRecord())
+{
+    $gMessage->show($gL10n->get('SYS_NO_ENTRY'));
+    // => EXIT
+}
+
+if(!$gCurrentUser->hasRightEditProfile($user))
+{
+    $gMessage->show($gL10n->get('SYS_NO_RIGHTS'));
+    // => EXIT
+}
+
+$sql = 'SELECT count(urt_id) FROM '.TBL_USER_RELATION_TYPES;
+$relationsStatement = $gDb->query($sql);
+if($relationsStatement->fetchColumn() == 0)
+{
+    $gMessage->show($gL10n->get('REL_NO_RELATION_TYPES_FOUND'));
+    // => EXIT
+}
+
+$headline = $gL10n->get('PRO_ADD_USER_RELATION');
+$gNavigation->addUrl(CURRENT_URL, $headline);
+
+// create html page object
+$page = new HtmlPage($headline);
+
+// add back link to module menu
+$relationEditMenu = $page->getMenu();
+$relationEditMenu->addItem('menu_item_back', $gNavigation->getPreviousUrl(), $gL10n->get('SYS_BACK'), 'back.png');
+
+// show form
+$form = new HtmlForm('relation_edit_form',
+                     $g_root_path.'/adm_program/modules/userrelations/userrelations_function.php?usr_id='.$getUsrId.'&amp;mode=1', $page);
+
+$form->addInput('usr_id', $gL10n->get('SYS_USER'), $user->getValue('FIRST_NAME').' '.$user->getValue('LAST_NAME'),
+         array('maxLength' => 100, 'property' => FIELD_DISABLED));
+
+// select box showing all relation types 
+$sql = 'SELECT urt_id, urt_name
+          FROM '.TBL_USER_RELATION_TYPES.'
+      ORDER BY urt_name';
+$form->addSelectBoxFromSql('urt_id', $gL10n->get('SYS_USER_RELATION'), $gDb, $sql, array('property' => FIELD_REQUIRED));
+
+// select box showing all users (TODO: check edit right)
+$sql = 'SELECT usr_id, concat(LAST_NAME.usd_value, \' \', FIRST_NAME.usd_value) AS name
+                  FROM '.TBL_MEMBERS.'
+            INNER JOIN '.TBL_ROLES.'
+                    ON rol_id = mem_rol_id
```

Comment:
You're right.

So in the first step we should only check `if(!$gCurrentUser->editUsers())` instead of `if(!$gCurrentUser->hasRightEditProfile($user))` like you do within **userrelations_function.php**.

The only users we miss are leaders with the right to edit users but this feature can be added later.